consensus/src/state_computer.rs

```diff
@@ -100,7 +100,10 @@ impl StateComputer for ExecutionProxy {
         // Similarily, after the state synchronization, we have to reset the cache
         // of BlockExecutor to guarantee the latest committed state is up to date.
         self.execution_correctness_client.lock().reset()?;
-        res?;
-        Ok(())
+
+        res.map_err(|error| {
+            let anyhow_error: anyhow::Error = error.into();
+            anyhow_error.into()
```

Comment:
Interesting, so you map an error into an `anyhow` and back into a `StateSyncError`?

Reply:
Yeah, there's a conversion from state_sync::error::Error into anyhow, and then StateSyncError in consensus has a conversion from anyhow. Unfortunately, the compiler can't do both jumps with a single `into` call 😄